### PR TITLE
add grommet 0.6.11 w/ npm auto-update

### LINKED ISSUE #7343: [Request] Add Grommet
**Library name:** Grommet
**Git repository url:** https://github.com/grommet/grommet
**npm package url(optional):** https://www.npmjs.com/package/grommet
**License(s):** Apache-2.0
**Official homepage:** http://grommet.io/

### PR DESCRIPTION
**Pull request by kennynaoh**
PR for #7343

@PeterDaveHello please help me review this pull request, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/grommet/grommet
 * Official website (optional, not the repository): http://grommet.io
 * NPM package url (optional): https://www.npmjs.com/package/grommet
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 3546
   - Count of watchers: 166
   - Count of forks: 2557
 * NPM download stats (optional):
   - Downloads in the last day: 54
   - Downloads in the last week: 805
   - Downloads in the last month: 2852

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [ ] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

autoupdate config added, close #7343, cc@alansouzati

### CODE REVIEW

File: ajax/libs/grommet/package.json [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  "homepage": "http://grommet.io",
+  "license": "Apache-2.0",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/grommet/grommet.git"
+  },
+  "bin": {
+    "grommet": "./bin/grommet.js"
+  },
+  "engine-strict": true,
```
Comment (PeterDaveHello) [inline ajax/libs/grommet/package.json]: Please remove `"bin", "engine-strict", "host" and "tmp"`

File: ajax/libs/grommet/0.5.5/grommet.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env node
```
Comment (PeterDaveHello) [inline ajax/libs/grommet/0.5.5/grommet.js]: It looks like a nodejs cli program, not a library runs in browser.

File: ajax/libs/grommet/package.json [hunk 3]
```diff
@@ -0,0 +1,107 @@
+{
+  "name": "grommet",
+  "filename": "grommet.min.js",
+  "npmName": "grommet",
+  "npmFileMap": [
+    {
+      "basePath": "bin",
```
Comment (PeterDaveHello) [inline ajax/libs/grommet/package.json]: What's the reason why to set `"bin"` as basePath? If you use it as basePath, how did you get the css files?

**Comment by kennynaoh** [comment]
Thanks for check my PR.
I have changed my file.
I found the Grommet.js file in grommet/components directory.
And after I do minify, it create .min.js and .min.js.map files
But After I pushed, I found that there are some files like .css or .js files in grommet/
I don't know whether I should put into the version file I created or not.
Please help me check again.
Thanks.

**Comment by eabay** [comment]
@kennynaoh Thanks for working on that!

Grommet has a [stable branch](https://github.com/grommet/grommet/tree/stable) in Github repo and it has dist versions of files. This may help.

**Comment by eabay** [comment]
https://github.com/cdnjs/cdnjs/issues/7344#issuecomment-202531740

This may help too.

**Comment by PeterDaveHello** [comment]
Thank you both @kennynaoh @eabay, please give me one more day to review this.

File: ajax/libs/grommet/package.json [hunk 4]
```diff
@@ -0,0 +1,101 @@
+{
+  "name": "grommet",
+  "filename": "Grommet.min.js",
+  "npmName": "grommet",
+  "npmFileMap": [
+    {
+      "basePath": "components",
+      "files": [
+        "**/*"
+      ]
```
Comment (PeterDaveHello) [inline ajax/libs/grommet/package.json]: There are so many files under its `components` folder, if you use `"**/*"` to include all the files, why did you only 3 files in your commit?

**Comment by PeterDaveHello** [comment]
@kennynaoh You should set the `basePath` to its root, and use `*grommet*.min.*` as filemap.

**Comment by kennynaoh** [comment]
@x09326 please help me check my PR
thanks

**Comment by kennynaoh** [comment]
I have updated the package.json about adding keywords.
Please review again.
Thanks.

**Comment by PeterDaveHello** [comment]
Something wrong in the commit message:
> autoupdate config added, close #7343, cc@alansouzati

**Comment by kennynaoh** [comment]
I have updated commit message already
Please review again
Thanks

**Comment by PeterDaveHello** [comment]
@kennynaoh you still need to wait for @x09326 's review, I'll tell you: In the commit message, please move the words about auto-update in the first line, the length is still enough, so that we can easily see it. Also in the PR title. You don't need to update it right now until @x09326 reviewed your PR.

**Comment by x09326** [comment]
I think this pr is okay, but I want to know why the license in the ```package.json``` is apache-2.0, but in the ```grommet.min.js``` they said they use MIT.

**Comment by PeterDaveHello** [comment]
Ping @kennynaoh, is there any updates? It's been a long time.

**Comment by PeterDaveHello** [comment]
@kennynaoh I think I answered your question there, please update this PR to v0.6.11 and let it been rebased ASAP.

**Comment by kennynaoh** [comment]
@x09326 
I have fixed the problem above.
Please help me review it again.
Thanks!

**Comment by PeterDaveHello** [comment]
@kennynaoh please don't forget the PR title.

**Pull request merged**